CMakeLists.txt

```
@@ -47,7 +36,19 @@ else()
 endif()
 
 ## Add serial library
+set(serial_HEADERS
+    include/serial/serial.h
+    include/serial/v8stdint.h
+)
+# Build, link and install main library
 add_library(${PROJECT_NAME} ${serial_SRCS})
```

```suggestion
add_library(serial ${serial_SRCS})
```
It aids in readability and `grep`ability if you simply spell out the variable name.

---

:+1: to this recommendation as it makes the cmake file much easier to read and understand. One of the hardest parts of cmake is reading the config files.

---

I agree here, but I'll wait for @wjwwood to chime in on this PR in general.

---

I am concerned this PR gets too big and goes the way of any of the other open PRs

---

It's fine with me, but it deviates from what it common in the ROS ecosystem.

---

> It's fine with me, but it deviates from what it common in the ROS ecosystem.

Not only the ROS ecosystem, but the whole cmake community I would say. For that reason I think this could be left as-is

---

@ChrisThrasher wrote

> https://github.com/wjwwood/serial/pull/285#discussion_r1231126932

No. This should IMHO be done like in [e1dabd8](https://github.com/wjwwood/serial/pull/231/commits/e1dabd86496113af3af5b84d385e5c64d869d5f7), taking into account whether to include v8stdint.h or not.